src/components/RiskImminentEventMap/index.tsx

```diff
@@ -0,0 +1,435 @@
+                mapOptions={defaultMapOptions}
+                navControlShown
+                navControlPosition="top-right"
+            >
+                {hazardKeys.map((key) => (
+                    <HazardMapImage
+                        key={key}
+                        hazardKey={key}
+                        onLoad={handleIconLoad}
+                    />
+                ))}
+                <MapContainer className={styles.mapContainer} />
+                <MapSource
+                    sourceKey="event-points"
+                    sourceOptions={geojsonSourceOptions}
+                    geoJson={pointFeatureCollection}
+                >
+                    {activePointFootprint && (
+                        <MapSource
+                            sourceKey="active-event-footprint"
+                            sourceOptions={geojsonSourceOptions}
+                            geoJson={activePointFootprint}
+                        >
+                            <MapLayer
+                                layerKey="exposure-fill"
+                                layerOptions={{
+                                    type: 'fill',
+                                    filter: [
+                                        '==',
+                                        ['get', 'type'],
+                                        'exposure',
+                                    ],
+                                    paint: {
+                                        'fill-color': COLOR_BLACK,
+                                        'fill-opacity': 0.2,
+                                    },
+                                }}
+                            />
+                            <MapLayer
+                                layerKey="track-outline"
+                                layerOptions={{
+                                    type: 'line',
+                                    filter: [
+                                        '==',
+                                        ['get', 'type'],
+                                        'track',
+                                    ],
+                                    paint: {
+                                        'line-color': COLOR_BLACK,
+                                        'line-opacity': 0.4,
+                                    },
+                                }}
+                            />
+                            <MapLayer
+                                layerKey="track-point"
+                                layerOptions={{
+                                    type: 'circle',
+                                    filter: [
+                                        '==',
+                                        ['get', 'type'],
+                                        'track-point',
+                                    ],
+                                    paint: {
+                                        'circle-radius': 4,
```

Comment:
Let's move the options outside the render.
This will call mapbox functions on each render.